tests/integration/interface_common_test.py

```diff
@@ -34,3 +44,59 @@ def test_iface_description_removal(eth1_up):
     libnmstate.apply(desired_state)
     current_state = statelib.show_only(('eth1',))
     assert Interface.DESCRIPTION not in current_state[Interface.KEY][0]
+
+
+def test_take_over_virtual_interface_then_remove():
+    assert _add_ip_link_dummy(DUMMY_INTERFACE) == 0
```

Comment:
This needs a contextmanager, for the same reason as mentioned in the next comment below.
It actually fits a fixture as it is part of the test setup (and teardown) and not something that the test itself cares to check.

Reply:
Done, thanks